lib/filter/QubitLimitIp.class.php

```diff
@@ -52,16 +56,52 @@ public function execute($filterChain)
         $filterChain->execute();
     }
 
-    protected function getRemoteAddress()
+    public function setLimit()
     {
-        $pathInfo = $this->request->getPathInfoArray();
+        if (null != sfConfig::get('app_limit_admin_ip')) {
+            $this->limit = explode(';', sfConfig::get('app_limit_admin_ip'));
+        } else {
+            $this->limit = [];
+        }
+    }
 
-        return $pathInfo['REMOTE_ADDR'];
+    /**
+     * Check if this request matches any exception rules.
+     * Returns true if a match is found, otherwise false.
+     *
+     * @param $debug
+     * @param $limit
+     * @param $firstCall
+     * @param $logout
+     */
+    public function shouldBypassLimit($debug, $limit, $firstCall, $logout)
```

Comment:
Why not `($debug, $limit, $firstCall, $module, $action)`? This might make validating the values simpler.

Reply:
good point, thanks @sbreker , i've made the changes!

Reply:
ps sorry if the rebased commits makes these changes harder to find in - i wanted to separate the QubitLimitIpFilter and QubitLimitIpFilterTest changes so that i could cherry pick QubitLimitIpFilter changes to analyst vm for testing SSO logout 😅